service/src/test/java/cz/muni/fi/pa165/mushrooms/service/facade/withMock_VisitFacadeImplTest.java

```diff
@@ -0,0 +1,306 @@
+package cz.muni.fi.pa165.mushrooms.service.facade;
+
+import cz.muni.fi.pa165.mushrooms.dto.*;
+import cz.muni.fi.pa165.mushrooms.entity.Forest;
+import cz.muni.fi.pa165.mushrooms.entity.Mushroom;
+import cz.muni.fi.pa165.mushrooms.entity.MushroomHunter;
+import cz.muni.fi.pa165.mushrooms.entity.Visit;
+import cz.muni.fi.pa165.mushrooms.enums.MushroomType;
+import cz.muni.fi.pa165.mushrooms.service.*;
+import cz.muni.fi.pa165.mushrooms.service.config.ServiceConfiguration;
+import mockit.Deencapsulation;
+import mockit.Delegate;
+import mockit.Expectations;
+import mockit.Injectable;
+import mockit.Tested;
+import org.dozer.Mapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+
+import javax.inject.Inject;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+/**
+ * Basic tests for hunterFacade implementations using a mock of the service layer.
+ *
+ * @author bkompis
+ */
+//TODO: ex throwing lower; DataAccessException (from hunterFacade impl)
+// TODO: constraints, exceptions in javadoc
+// TODO: equals()in DTOs vs. equals on all attributes
+// TODO: exception handling
+// TODO: more tests, + tests without mock (integration)
+@ContextConfiguration(classes = ServiceConfiguration.class)
+@TestExecutionListeners(TransactionalTestExecutionListener.class) // TODO: necessary?
+public class withMock_VisitFacadeImplTest extends AbstractTransactionalJUnit4SpringContextTests { //should be transactional?
+
+    //private final Logger logger = LoggerFactory.getLogger(MushroomHunterFacadeImplTest_withMock.class);
+
+    private MushroomHunter hunter1;
+    private MushroomHunter hunter2;
+
+    private MushroomHunterDTO hunter1DTO;
+    private MushroomHunterDTO hunter2DTO;
+
+    private Forest forest1;
+    private Forest forest2;
+
+    private ForestDTO forest1DTO;
+    private ForestDTO forest2DTO;
+
+    private Mushroom mushroom1;
+    private Mushroom mushroom2;
+
+    private MushroomDTO mushroom1DTO;
+    private MushroomDTO mushroom2DTO;
+
+    private Visit visit1;
+    private Visit visit2;
+
+    private VisitDTO visit1DTO;
+    private VisitDTO visit2DTO;
+
+    @Injectable
+    private MushroomHunterService service;
+
+    @Injectable
+    private ForestService forestService;
+
+    @Injectable
+    private MushroomService mushroomService;
+
+    @Injectable
+    private VisitService visitService; // ak je ma sa injectnut
```

Comment:
@Injectable creates a mock with default implementations of methods (returning default type value), is that what you want?